Network/DNS/Resolver.hs

```diff
@@ -342,15 +369,37 @@ lookupRawInternal rcv ad rlv dom typ = do
               Nothing  -> loop query checkSeqno (cnt + 1) False
               Just res -> do
                   let valid = checkSeqno res
-                  if valid then
-                      return $ Right res
-                    else
-                      loop query checkSeqno (cnt + 1) False
+                  case valid of
+                      False  -> loop query checkSeqno (cnt + 1) False
+                      True | False <- trunCation $ flags $ header res
```

Comment:
This is not readable for me.
I guess that `trunCation $ flags $ header res` should be bound a variable in the `let` above and use `not var`.